app/src/main/java/org/fossasia/phimpme/trashbin/TrashBinAdapter.java

```diff
@@ -0,0 +1,126 @@
+        trashItemsList = list;
+    }
+
+    @Override public TrashBinAdapter.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+        View view = LayoutInflater.from(parent.getContext())
+                .inflate(R.layout.trashbin_item_view, null, false);
+        view.setLayoutParams(new RecyclerView.LayoutParams(RecyclerView.LayoutParams.MATCH_PARENT,
+                RecyclerView.LayoutParams.WRAP_CONTENT));
+        //view.setOnClickListener(onClickListener);
+        return new TrashBinAdapter.ViewHolder(view);
+    }
+
+    @Override public void onBindViewHolder(final TrashBinAdapter.ViewHolder holder, int position) {
+
+        if(trashItemsList.size() != 0){
+            TrashBinRealmModel trashBinRealmModel = trashItemsList.get(position);
+            String date = trashBinRealmModel.getDatetime();
+            try {
+                DateFormat format = new SimpleDateFormat("dd/MM/yyyy hh:mm:ss");
+                Date parsedDate = format.parse(date);
+                DateFormat uploadDate = new SimpleDateFormat("dd/MM/yyyy");
+                DateFormat uploadTime = new SimpleDateFormat("hh:mm:ss");
+                holder.deleteDate.setText(uploadDate.format(parsedDate));
+                holder.deleteTime.setText(uploadTime.format(parsedDate));
+            } catch (ParseException e) {
+                e.printStackTrace();
+            }
+            holder.deleteDate.setTag(trashBinRealmModel);
+            Uri uri = Uri.fromFile(new File(trashBinRealmModel.getTrashbinpath()));
+            Glide.with(getApplicationContext()).load(uri)
+                    .diskCacheStrategy(DiskCacheStrategy.ALL)
+                    .into(holder.deletedImage);
+            holder.popupMenuButton.setOnClickListener(new View.OnClickListener() {
+                @Override public void onClick(View view) {
+                    PopupMenu menu = new PopupMenu(getApplicationContext(), holder.popupMenuButton);
+                    menu.inflate(R.menu.menu_popup_trashbin);
+                    menu.setOnMenuItemClickListener(new PopupMenu.OnMenuItemClickListener() {
+                        @Override public boolean onMenuItemClick(MenuItem menuItem) {
+                            switch (menuItem.getItemId()){
+
+                                case R.id.restore_option:
+                                    return true;
+
+                                case R.id.delete_permanently:
+                                    return true;
+
+                                default:
+                                    return false;
+
+                            }
+                        }
+                    });
+                    menu.show();
+                }
+            });
+            //holder.popupMenuButton.setImageResource(R.drawable.ic_more_vert_black_24dp);
+        }
+    }
+
+    public void setResults(ArrayList<TrashBinRealmModel> results) {
+        trashItemsList = results;
+        notifyDataSetChanged();
```

Comment:
This operation is extremely expensive if used multiple times. Please use the `DiffUtil`. https://developer.android.com/reference/android/support/v7/util/DiffUtil